### PR TITLE
Refactor dockerconfig (ex-jsonconfig)

### PR DESCRIPTION
**Pull request by ivanilves**

### CODE REVIEW

**Comment by ivanilves** [comment]
This is a PR I need to merge to make one change for "nois macos i suaus".
It was made in train this morning. I only merge it during my working hours.

File: docker/config/config.go [hunk 1]
```diff
@@ -28,12 +40,21 @@ func (c *Config) IsEmpty() bool {
...
 	}
 
 	return c.usernames[registry], c.passwords[registry], true
 }
 
+// GetRegistryAuth gets per-registry base64 authentication string
+func (c *Config) GetRegistryAuth(registry string) (string, bool) {
+	username, password, defined := c.GetCredentials(registry)
+
+	jsonString := fmt.Sprintf("{ \"username\": \"%s\", \"password\": \"%s\" }", username, password)
```
Comment (Fsero) [inline docker/config/config.go]: maybe you can use `{ "username": "%s", "password": "%s" }` to avoid some slashes?

Comment (ivanilves) [reply docker/config/config.go]: ups sorry @Fsero but I can not get what do you mean :cry

Comment (ivanilves) [reply docker/config/config.go]: I need to merge this so one thing I need to push for nois macos would be unblocked.

**Pull request merged**